aws/request.go

```diff
@@ -217,3 +227,87 @@ func (r *Request) Send() error {
 
 	return nil
 }
+
+func (r *Request) HasNextPage() bool {
+	return r.nextPageToken() != nil
+}
+
+func (r *Request) nextPageToken() interface{} {
+	if r.Operation.Paginator == nil {
+		return nil
+	}
+	v := awsutil.ValuesAtAnyPath(r.Data, r.Operation.OutputToken)
+	if v != nil && len(v) > 0 {
+		return v[0]
+	}
+	return nil
+}
+
+func (r *Request) NextPage() *Request {
+	token := r.nextPageToken()
+	if token == nil {
+		return nil
+	}
+
+	if r.Operation.TruncationToken != "" {
+		tr := awsutil.ValuesAtAnyPath(r.Data, r.Operation.TruncationToken)
+		if tr == nil {
+			return nil
+		} else if len(tr) > 0 {
+			switch v := tr[0].(type) {
+			case bool:
+				if v == false {
+					return nil
+				}
+			}
+		}
+	}
+
+	data := reflect.New(reflect.TypeOf(r.Data).Elem()).Interface()
+	nr := NewRequest(r.Service, r.Operation, awsutil.CopyOf(r.Params), data)
+	awsutil.SetValueAtAnyPath(nr.Params, nr.Operation.InputToken, token)
+	return nr
+}
+
+// This nil value is needed to work around passing nil into reflect.Value#Call()
+// See https://groups.google.com/forum/#!topic/golang-nuts/apNcACpl_fI
+var nilval interface{}
+var rnil = reflect.ValueOf(&nilval).Elem()
+
+func (r *Request) EachPage(fn interface{}) error {
+	valfn := reflect.ValueOf(fn)
+	if valfn.Kind() != reflect.Func {
+		panic("expected function for EachPage()")
```

Comment:
Not passing a function to this method would be a programmer error, not some kind of runtime error. It would never make sense to handle an error like that, since there would be no sensible recovery. Basically this *should* be a compile-time error but Go cannot quite represent the type here. That said, `EachPage()` is a low-level call that is typically not used unless you know what you're doing. The `*Pages()` methods have the correct type checking in them.